src/core/Link/Link.tsx

```diff
@@ -0,0 +1,31 @@
+import React from 'react';
+import { NavLink } from 'react-router-dom';
+import { useStyles } from './styles';
+
+interface LinkProps {
+  pathname: string;
+  disabled?: boolean;
+  target?: string;
+}
+
+const Link: React.FC<LinkProps> = ({
+  pathname,
+  target,
+  children,
+  disabled,
+}) => {
+  const classes = useStyles();
+  const disableLink = disabled ? classes.disabled : '';
+  return (
+    <NavLink
+      className={`${classes.root} ${disableLink}`}
+      to={pathname}
+      target={target}
```

Comment:
Line no 22 and 23 will not be required then

Reply:
@arkajyotiMukherjee 
'to' and 'target' are being passed via props. That is, the logic is correct here. I can remove the 'target', but I guess 'to' is needed.

Reply:
to and target will automatically come from LinkBaseProps...take a look at the other changes...I've provided the code for all the changes @Hasmik8